### PR TITLE
Add extra api methods to PlanDefinitionProcessor and ActivityDefinitiionProcessor

### PR DESCRIPTION
**Pull request by ktarasenko**
Allow PlanDefinitionProcessor and ActivityDefinitiionProcessor to evaluate PlanDefinitions and ActivityDefinitions by id, url and directly passing the object

### CODE REVIEW

File: evaluator.activitydefinition/src/main/java/org/opencds/cqf/cql/evaluator/activitydefinition/BaseActivityDefinitionProcessor.java [hunk 1]
```diff
@@ -63,16 +67,67 @@ public IBaseResource apply(IIdType theId, String subjectId, String encounterId,
...
-    this.libraryEngine = libraryEngine;
     T activityDefinition = (T) this.repository.read(ActivityDefinition.class, theId);
     if (activityDefinition == null) {
       throw new IllegalArgumentException("Couldn't find ActivityDefinition " + theId);
     }
+    return apply(activityDefinition, subjectId, encounterId, practitionerId, organizationId,
+        userType, userLanguage, userTaskContext, setting, settingContext, parameters, libraryEngine);
+  }
+
+  public IBaseResource apply(String url, String subjectId, String encounterId,
```
Comment (ktarasenko) [inline evaluator.activitydefinition/src/main/java/org/opencds/cqf/cql/evaluator/activitydefinition/BaseActivityDefinitionProcessor.java:78]: @JPercival  What is the good type to use for Url?

The `IPrimitiveType<String>` collides with `IIdType` and actual `CanonicalType`/`UrlType` is version specific

**Comment by JPercival** [comment]
@barhodes - Can you take a look at this?

**Review by barhodes — COMMENTED** [review]
Canonical and Resource parameters have been added to $apply and other clinical reasoning operations.  This PR should no longer be necessary.

**Comment by JPercival** [comment]
@ktarasenko - Can you take a look at the latest code on `master` and see if your needs are addressed?

**Comment by ktarasenko** [comment]
Yes, that solves it. Great timing @barhodes! 😄

**Pull request closed without merging**